sim/dex.ts

```diff
@@ -1265,7 +1278,8 @@ class ModdedDex {
 				return new TypeError(`${filePath}, if it exists, must export a non-null object`);
 			}
 			if (!dataObject[key] || typeof dataObject[key] !== 'object') {
-				return new TypeError(`${filePath}, if it exists, must export an object whose '${key}' property is a non-null object`);
+				return new TypeError(
+					`${filePath}, if it exists, must export an object whose '${key}' property is a non-null object`);
```

Comment:
I would whitelist `return new TypeError` in the long line regex.

Reply:
Its kind of a dense regex, so I'll maybe let you edit that one...

FWIW, I'm also not sure I'd want line length exceptions to be whitelisted - I think there should be so few places that breaking the line limit should be necessary that manual overrides should be sufficient (remember, I'm from a world where the automatic formatter decides how long my lines are and where they wrap!).

*(In news somewhat related to this change, I've come around and agree with you that I don't like what the publicly availably automatic code formatting options do to the PS! codebase, see https://github.com/Zarel/Pokemon-Showdown/compare/master...pkmn-cc:prettier if you want to see how much **less** prettier things become...)*